masq_lib/src/test_utils/environment_guard.rs

```diff
@@ -43,16 +42,45 @@ impl<'a> Drop for EnvironmentGuard<'a> {
         self.environment
             .iter()
             .for_each(|(name, value)| std::env::set_var(name, value));
+        let mut thread_id_guard = ENVIRONMENT_GUARD_THREAD_ID
```

Comment:
Enhanced `EnvironmentGuard` with a thread ID so that A) if we couldn't lock the guard we could produce a message saying who had it, B) we could tell if our thread already had the `EnvironmentGuard` locked and panic (instead of deadlock) if we tried to lock it again, and C) when we went to read or write the main and alias static `CryptDE`s, we could check to see if the `EnvironmentGuard` was owned by our thread and panic if it wasn't--that is, make tests that interact with static `CryptDE`s fail if they didn't lock the `EnvironmentGuard` first.

So...eventually we decided that static `CryptDE`s were a bad idea and got rid of them, but we're leaving the thread ID in the `EnvironmentGuard` anyway, for the first two reasons.